src/reducers/authReducer.js

```diff
@@ -0,0 +1,78 @@
+import {
+  LOGIN_REQUEST,
+  LOGIN_SUCCESS,
+  LOGIN_FAILURE,
+  SIGNUP_REQUEST,
+  SIGNUP_SUCCESS,
+  SIGNUP_FAILURE,
+  LOGOUT_USER,
+  WHOAMI_REQUEST,
+  WHOAMI_SUCCESS,
+  WHOAMI_FAILURE,
+  REDIRECT
+} from "../actions/constants";
+
+import initialState from "./defaultState";
+
+
+export default function auth(state = initialState, action: Object) {
+  switch (action.type) {
+    case LOGIN_REQUEST:
+    case WHOAMI_REQUEST:
+    case SIGNUP_REQUEST:
+      return state;
+
+    case LOGIN_SUCCESS:
+      state = Object.assign({}, state, {
+        authenticated: true
+      })
+      return state
+
+    case LOGIN_FAILURE:
+      return Object.assign({}, state, {
+        authenticated: false,
+        message: action.payload.message,
+        status: action.payload.status
+      })
+
+    case LOGOUT_USER:
+      localStorage.removeItem("awesome_bucketlist_token");
```

Comment:
By convention, reducers aren't allowed to have side-effects. Let's not perform localStorage manipulation in this reducer.